firstmicroservice/src/main/java/com/stgcodes/validation/AddressValidator.java

```diff
@@ -38,23 +39,23 @@ private void validateLineTwo(Address address, Errors errors) {
         String lineTwo = address.getLineTwo().replaceAll(WHITESPACE_DASH_SLASH_MATCHER, "");
 
         if (lineTwo.length() > 25 || lineTwo.length() == 0) {
-            errors.rejectValue("linetwo", "linetwo.format");
+            errors.rejectValue("lineTwo", "linetwo.format");
         }
     }
 
     private void validateCity(Address address, Errors errors) {
         String city = address.getCity().replaceAll(WHITESPACE_DASH_SLASH_MATCHER, "");
 
-        if (city.length() > 50 || city.length() == 0) {
+        if (city.length() > 21 || city.length() == 0) {
             errors.rejectValue("city", "city.format");
         }
     }
 
     private void validateState(Address address, Errors errors) {
-        String state = address.getState().replaceAll(WHITESPACE_DASH_SLASH_MATCHER, "");
+        String state = address.getState().trim();
 
-        if (state.length() > 13 || state.length() == 0) {
-            errors.rejectValue("state", "state.format");
+        if(GeographicState.valueOfAbbreviation(state) == null && GeographicState.valueOfName(state) == null) {
```

Comment:
There's probably a better way to handle this.  Write a method in the GeographicState enum that matches the value passed in and return true/false.  Maybe something like a contains method that a java.util.List uses.